src/internal/cache-client.ts

```diff
@@ -258,6 +261,127 @@ export class CacheClient {
     });
   }
 
+  public async listFetch(
+    cacheName: string,
+    listName: string | Uint8Array
+  ): Promise<CacheListFetch.Response> {
+    try {
+      validateCacheName(cacheName);
+      validateListName(listName);
+    } catch (err) {
+      return new CacheListFetch.Error(normalizeSdkError(err as Error));
+    }
+    this.logger.trace(
+      `Issuing 'listFetch' request; listName: ${listName.toString()}`
+    );
+    const result = await this.sendListFetch(cacheName, this.convert(listName));
+    this.logger.trace(`'listFetch' request result: ${result.toString()}`);
+    return result;
+  }
+
+  private async sendListFetch(
+    cacheName: string,
+    listName: Uint8Array
+  ): Promise<CacheListFetch.Response> {
+    const request = new cache.cache_client._ListFetchRequest({
+      list_name: listName,
+    });
+    const metadata = this.createMetadata(cacheName);
+
+    return await new Promise(resolve => {
+      this.clientWrapper.getClient().ListFetch(
+        request,
+        metadata,
+        {
+          interceptors: this.interceptors,
+        },
+        (err, resp) => {
+          if (resp) {
+            if (resp.missing) {
+              resolve(new CacheListFetch.Miss());
+            } else if (resp.found) {
+              resolve(new CacheListFetch.Hit(resp.found.values));
+            } else {
+              resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
+            }
+          } else {
+            resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
```

Comment:
I think we can use switch like [this](https://github.com/momentohq/client-sdk-javascript/blob/main/src/internal/cache-client.ts#L232)?

Reply:
I couldn't figure out how to make it work. The other response has a single property to check against an enum. This one has multiple boolean properties.

Reply:
How about like this?
```
// enum needs to be outside of CacheClient class
enum DataStructureResponse {
  None = 'none',
  Found = 'found',
  Missing = 'missing',
}

(err, resp) => {
          if (resp) {
            switch (resp.list) {
              case DataStructureResponse.Missing:
                resolve(new CacheListFetch.Miss());
                break;
              case DataStructureResponse.Found:
                resolve(new CacheListFetch.Hit(resp.found.values));
                break;
              default:
                resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
                break;
            }
          } else {
            resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
          }
        }
```

Reply:
That'll work. However, it's a lot of extra scaffolding for the same code. It turns the original protobuf number into a string and back into a number just to use a switch.

Reply:
Do we need to do that?
> It turns the original protobuf number into a string and back into a number just to use a switch.

I thought `resp.list` returns string (`missing`, `found`, `none`).
Sorry if I misunderstood your concern.

Reply:
At least that's what my VS code tells me 🤷‍♀️ 
![image](https://user-images.githubusercontent.com/22686260/212170276-02dab76b-52b4-449f-ba85-786435955d74.png)

Reply:
But I agree that if it's too much work to make this a switch statement, then it's fine to use a if statement here 👍

Reply:
I tried this.

```
        (err, resp) => {
          switch (resp) {
            case resp.missing instanceof cache_client._ListFetchResponse._Missing:
              resolve(new CacheListFetch.Miss());
              break;
            case resp.found instanceof cache_client._ListFetchResponse._Found:
              resolve(new CacheListFetch.Hit(resp.found.values));
              break;
            default:
              resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
              break;
          }
        }
```

Which *should* work, but eslint doesn't recognize that `resp.found` is a _Found object.

```
  303:46  error  Unsafe argument of type `any` assigned to a parameter of type `Uint8Array[]`  @typescript-eslint/no-unsafe-argument
  303:46  error  Unsafe member access .values on an `any` value                                @typescript-eslint/no-unsafe-member-access
```

Reply:
```
        (err, resp) => {
          switch (true) {
            case typeof resp === 'undefined':
            default:
              resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
              break;
            case resp.missing instanceof
              cache_client._ListFetchResponse._Missing:
              resolve(new CacheListFetch.Miss());
              break;
            case resp.found instanceof cache_client._ListFetchResponse._Found:
              resolve(new CacheListFetch.Hit(resp.found.values));
              break;
          }
        }
```

Doesn't work either, eslint thinks resp can be undefined.

I have to stop obsessing over this, so I asked StackOverflow.
https://stackoverflow.com/questions/75102169/how-do-i-narrow-a-value-to-not-be-undefined

Reply:
They came through. https://stackoverflow.com/a/75102198/14660

```
        (err, resp) => {
          if (resp?.missing) {
            resolve(new CacheListFetch.Miss());
          } else if (resp?.found) {
            resolve(new CacheListFetch.Hit(resp.found.values));
          } else {
            resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
          }
        }
```

For some reason this doesn't work as a switch. 🤷

Reply:
That works. We can check more directly...

```
          switch (resp?.list) {
            case 'missing':
              resolve(new CacheListFetch.Miss());
              break;
            case 'found':
              resolve(new CacheListFetch.Hit(resp.found.values));
              break;
            default:
              resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
              break;
          }
```

Reply:
Can we use enum so that we can use missing/found in other data structure APIs uniformly?
Something like this?
```
enum DataStructureResponse {
  None = 'none',
  Found = 'found',
  Missing = 'missing',
}
```

Reply:
Then we're back to adding a lot of extra code just to use a switch.

I think the if/elseif/else here is fine. As long as its DRY.